kernel/src/engine/parquet_stats_skipping.rs

```diff
@@ -1,14 +1,94 @@
 //! An implementation of data skipping that leverages parquet stats from the file footer.
 use crate::expressions::{
-    BinaryOperator, ColumnName, Expression, Scalar, UnaryOperator, VariadicOperator,
+    BinaryOperator, ColumnName, Expression as Expr, Scalar, UnaryOperator, VariadicOperator,
+};
+use crate::predicates::{
+    DataSkippingPredicateEvaluator, PredicateEvaluator, PredicateEvaluatorDefaults,
 };
 use crate::schema::DataType;
 use std::cmp::Ordering;
-use tracing::info;
 
 #[cfg(test)]
 mod tests;
 
+/// A helper trait (mostly exposed for testing). It provides the four stats getters needed by
+/// [`DataSkippingStatsProvider`]. From there, we can automatically derive a
+/// [`DataSkippingPredicateEvaluator`].
+pub(crate) trait ParquetStatsProvider {
+    fn get_parquet_min_stat(&self, _col: &ColumnName, _data_type: &DataType) -> Option<Scalar>;
+
+    fn get_parquet_max_stat(&self, _col: &ColumnName, _data_type: &DataType) -> Option<Scalar>;
+
+    fn get_parquet_nullcount_stat(&self, _col: &ColumnName) -> Option<i64>;
+
+    fn get_parquet_rowcount_stat(&self) -> i64;
```

Comment:
nit: should we move over the doc comments on each? e.g. `/// Retrieves the minimum value of a column, if it exists and has the requested type.`